### PR TITLE
Defect/de434 check giving decimal fix

### PR DESCRIPTION
**Pull request by IngageRitterS**
Allow for checks to process with dollars and cents
Make sure all emails generate with the correctly formatted amount
Remove the space between the exclamation point and the fund.

### CODE REVIEW

File: crossroads.net/app/give/one_time/templates/one_time_thank_you.html [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <div class="form-group">
   <div class="push-top push-bottom">
       <h4>Thank You</h4>
-      <p>Thank you for your <strong>{{give.dto.amount | currency}}</strong> gift to <strong>{{give.dto.program['Name']}}</strong>
+      <p>Thank you for your <strong>{{give.dto.amount | currency}}</strong> gift to <strong>{{give.dto.program['Name']}}!</strong>
         <span ng-if="give.dto.mockPledge">
             <!-- TEMPORARY - if this is towards a specific commitment, we want to put the commitment owner's full name here to make it clear who it's going to -->
             towards <strong>First Last's</strong> commitment
-        </span>!
```
Comment (IngageKrizJ) [inline crossroads.net/app/give/one_time/templates/one_time_thank_you.html]: Sandi, that space was bothering me too, however, this change would end up showing "Thank you for your $123 gift to Crossroads! towards Jim Kriz's commitment" if this was for a commitment...  I'm not sure if we can get the exclamation int the right place without the extra space without a bunch if ng-if/ng-show tags.

Comment (IngageRitterS) [inline crossroads.net/app/give/one_time/templates/one_time_thank_you.html]: BOO

On Tue, Nov 24, 2015 at 2:23 PM, Jim Kriz <notifications@github.com> wrote:

> In crossroads.net/app/give/one_time/templates/one_time_thank_you.html
> <https://github.com/crdschurch/crds-angular/pull/1192#discussion_r45781504>
> :
>
> >          <span ng-if="give.dto.mockPledge">
> >              
> >              towards <strong>First Last's</strong> commitment
> > -        </span>!
>
> Sandi, that space was bothering me too, however, this change would end up
> showing "Thank you for your $123 gift to Crossroads! towards Jim Kriz's
> commitment" if this was for a commitment... I'm not sure if we can get the
> exclamation int the right place without the extra space without a bunch if
> ng-if/ng-show tags.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/crdschurch/crds-angular/pull/1192/files#r45781504>.
>



-- 



[image: Logo: Ingage Partners, Consulting with Purpose]
<http://www.ingagepartners.com/>

Sandi Ritter
Senior Consultant
513.321.7700 (w) *|* 859.801.6028 (m)
2753 Observatory Ave *|* Cincinnati, OH 45208
www.IngagePartners.com <http://www.ingagepartners.com/>
[image: B-Corp Button] <http://www.bcorporation.net/ingagepartners> [image:
Linked-in Button] <http://www.linkedin.com/company/ingage-partners-llc> [image:
Facebook Button]
<http://www.facebook.com/pages/Ingage-Partners/337211479674920>

File: Gateway/crds-angular/Services/EzScanCheckScannerService.cs [hunk 2]
```diff
@@ -64,14 +65,8 @@ public CheckScannerBatch CreateDonationsForBatch(CheckScannerBatch batchDetails)
...
                     StripeCharge charge;
-                    if (contactDonor.HasAccount)
-                    {
-                        charge = _paymentService.ChargeCustomer(contactDonor.ProcessorId, contactDonor.Account.ProcessorAccountId, (int) (check.Amount), contactDonor.DonorId);
-                    }
-                    else
-                    {
-                        charge = _paymentService.ChargeCustomer(contactDonor.ProcessorId, (int)(check.Amount), contactDonor.DonorId);   
-                    }
+                    var decimalAmt = check.Amount * Constants.StripeDecimalConversionValue;
```
Comment (dtkocher) [inline Gateway/crds-angular/Services/EzScanCheckScannerService.cs]: Remove this.

File: Gateway/crds-angular/Services/StripeService.cs [hunk 3]
```diff
@@ -268,7 +268,7 @@ private static SourceData MapDefaultSource(List<SourceData>sources, string defau
         public StripeCharge ChargeCustomer(string customerToken, int amount, int donorId)
         {
             var request = new RestRequest("charges", Method.POST);
-            request.AddParameter("amount", amount * Constants.StripeDecimalConversionValue);
```
Comment (dtkocher) [inline Gateway/crds-angular/Services/StripeService.cs]: Add this back in.  The ChargeCustomer is being called when processing online gifts and therefore not converting to cents.

**Pull request merged**